pandas/core/arrays/base.py

```diff
@@ -1207,7 +1214,7 @@ def _formatter(self, boxed: bool = False) -> Callable[[Any], str | None]:
     # Reshaping
     # ------------------------------------------------------------------------
 
-    def transpose(self, *axes) -> ExtensionArray:
+    def transpose(self, *axes: int) -> ExtensionArray:
```

Comment:
If this is for compatibility with numpy, `int` is not correct according to the `np.ndarray.transpose` docstring (None or tuple of ints)

Reply:
I'm changing this to be `axes: NumpyAxis` to be compatible with numpy where `NumpyAxis` is `NumpyAxis = Union[Tuple[int], List[int], None]`